src/latch_cli/nextflow/config.py

```diff
@@ -1,57 +1,235 @@
 import json
+import re
+from dataclasses import MISSING, Field, field, make_dataclass
+from enum import Enum
 from pathlib import Path
 from textwrap import dedent
-from typing import Optional, Type, Union
+from typing import Annotated, Optional, TypeVar, Union
 
 import click
+import google.protobuf.json_format as gpjson
+from flytekit.core.annotation import FlyteAnnotation
+from flytekit.core.context_manager import FlyteContext, FlyteContextManager
+from flytekit.core.type_engine import TypeEngine
 
-from latch.types.directory import LatchDir, LatchOutputDir
+from latch.ldata.path import LPath
+from latch.types.directory import LatchDir
 from latch.types.file import LatchFile
-from latch_cli.snakemake.config.utils import type_repr
-from latch_cli.snakemake.utils import reindent
-
-
-def get_param_type(
-    details: dict,
-) -> Union[
-    Type[str], Type[bool], Type[int], Type[float], Type["LatchFile"], Type["LatchDir"]
-]:
-    t = details.get("type")
-    if t is None:
-        # rahul: assume string if type is not specified
-        return str
+from latch.types.samplesheet_item import SamplesheetItem
+from latch_cli.snakemake.config.utils import dataclass_repr, get_preamble
+from latch_cli.utils import best_effort_display_name, identifier_from_str
+
+from .parse_schema import NfType, parse_schema
+
+underscores = re.compile(r"_+")
+
+
+def best_effort_title_case(s: str) -> str:
+    return identifier_from_str("".join(underscores.sub(" ", s).title().split()))
+
+
+T = TypeVar("T")
+
+
+def as_python_val(typ: type[T], value: object) -> T:
+    try:
+        return typ(value)
+    except:
+        return value
+
+
+def get_field_type(
+    ctx: FlyteContext, idx: int, name: str, props: NfType
+) -> tuple[str, type, Field[object]]:
+    if not name.isidentifier():
+        name = f"param_{idx}"
 
-    if t == "string":
-        format = details.get("format")
-        if format is not None:
-            if format == "file-path":
-                return LatchFile
-            elif format == "directory-path":
-                return LatchDir
+    inner = get_python_type_inner(name, props)
+    python_typ = inner if props["metadata"]["required"] else Optional[inner]
```

Comment:
use `get_python_type` here?

Reply:
i use the inner type in certain places below